website/docs/configuration.mdx

```diff
@@ -49,6 +54,14 @@ module.exports = {
 };
 ```
 
+:::info[About the default `docusaurus.config.js` template]
+
+The init template `docusaurus.config.js` uses the `const config = { ... }` and it is equivalent to `module.exports = { ... }` shown throughout the examples.
+There are various equivalent ways to declare your Docusaurus config, such as using creator functions and async code.
+See: [**`docusaurus.config.js` API reference**](api/docusaurus.config.js.mdx)
+
+:::
+
```

Comment:
This is in the `### Theme, plugin, and preset configurations` section, and does not look like an appropriate place to me

I'd suggest to create a paragraph `## Syntax to declare docusaurus.config.js`, and add more content here, including multiple non-inline code examples, including async examples currently found in the API ref, link to nodejs doc about import/require etc

Reply:
Which doc link is more important for nodejs import/require?  I ask because is `docusaurus.config.js` expecting only CommonJS or utilizing ECMAScript module loading and has interoperability with CommonJS?
https://nodejs.org/api/esm.html#modules-ecmascript-modules
and
https://nodejs.org/api/esm.html#interoperability-with-commonjs

I see the compiler target itself is ES2020?
```json
{
  "compilerOptions": {
    /* Emit */
    "target": "ES2020",
```

Reply:
We don't support ES modules, so my mistake we shouldn't link to any doc mentioning imports or this will be confusing. (including the Node.js docs unfortunately, hard to follow, not very well presented and also mentioning ES modules)

It is surprisingly hard to find easy-to-understand explanations of CommonJS modules. 
This one is pretty simple: https://flaviocopes.com/commonjs/

You can say something like:

> The Docusaurus config file only supports the [CommonJS](https://flaviocopes.com/commonjs/) module system (`require("lib")` / `module.exports = /* your config*/`).